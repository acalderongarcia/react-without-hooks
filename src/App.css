*{
  font-family: 'Roboto', sans-serif;
}

.App {
  text-align: center;
}

.firstSection{
  background-color:transparent !important;
}

.secondSection{
  background-color:rgb(243, 242, 242) !important;
}

.logo{
  height: 325px;
}

.downloadApple{
  height: 50px;
}

.blue-text{
  color: rgb(28, 69, 255);
}

.btn{
  border-radius: 0!important;
}

.btn-blue{
  background-color: #1976d2!important;
  color: #fff!important;
}
