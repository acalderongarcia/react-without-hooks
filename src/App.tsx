import React from 'reactn';
import './App.css';
import 'bootstrap/dist/css/bootstrap.css';
import Header from "./components/Header";
import PropTypes from "prop-types";
import t from "./utils/localization/Localization";

class App extends React.Component{

    static propTypes = {
        children: PropTypes.object.isRequired
    };

    render() {
        t.setLanguage(this.global.currentLanguage)
        const children = this.props;
        return (
            <div>
                <Header/>
                {children.children}

                <body> 
                sssFFFFFFFFFFFFFFFFFF
                
                </body>
            </div>
        );
    }
}

export default App;
