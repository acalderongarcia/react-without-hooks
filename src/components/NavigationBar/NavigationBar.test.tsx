import React from 'reactn';
import renderer from 'react-test-renderer';
import '../../App.css'
import 'bootstrap/dist/css/bootstrap.css';
import { NavigationBar } from ".";
import t from '../../utils/localization/Localization';
import ReactFlagsSelect from 'react-flags-select';
import 'react-flags-select/css/react-flags-select.css';


describe('Testing src/components/NavigationBar/index.tsx', () => {

    test('Button it renders', () => {
        // Before
        const wrapper = renderer.create(

            <button className="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
                <span className="navbar-toggler-icon"></span>
            </button>

        );
        // Then
        let tree = wrapper.toJSON();
        // After
        expect(tree).toMatchSnapshot();
    });

    test('ReactFlagsSelect it renders', () => {
        // Before
        const wrapper = renderer.create(
            <div className="NavigationBar">
                <nav className="navbar navbar-expand-lg navbar-light bg-light">
                    <a className="navbar-brand" href="#">{t.title}</a>
                    <ReactFlagsSelect
                        onSelect={onLinkClick}
                        defaultCountry={t.getLanguage().toUpperCase()}
                        countries={["GB", "ES"]}
                        customLabels={{ "GB": "EN", "ES": "ES" }}
                     />
                </nav>
            </div>
        );
        // Then
        let tree = wrapper.toJSON();
        // After
        expect(tree).toMatchSnapshot();
    });

});

function onLinkClick(item: string) {
    let languageSelected = NavigationBar.prototype.global.currentLanguage;
    switch (item.toString()) {
        case "ES":
            languageSelected = "es";
            t.setLanguage("es")
            break;
        case "GB":
            languageSelected = "en";
            t.setLanguage("en")
            break;
    }
    NavigationBar.prototype.setGlobal({
        currentLanguage: languageSelected
    });


}


