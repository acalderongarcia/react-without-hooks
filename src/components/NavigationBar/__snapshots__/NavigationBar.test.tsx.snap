// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing src/components/NavigationBar/index.tsx Button it renders 1`] = `
<button
  aria-controls="navbarSupportedContent"
  aria-expanded="false"
  aria-label="Toggle navigation"
  className="navbar-toggler"
  data-target="#navbarSupportedContent"
  data-toggle="collapse"
  type="button"
>
  <span
    className="navbar-toggler-icon"
  />
</button>
`;

exports[`Testing src/components/NavigationBar/index.tsx ReactFlagsSelect it renders 1`] = `
<div
  className="NavigationBar"
>
  <nav
    className="navbar navbar-expand-lg navbar-light bg-light"
  >
    <a
      className="navbar-brand"
      href="#"
    >
      Title
    </a>
    <div
      className="flag-select "
    >
      <div
        className="selected--flag--option "
        onClick={[Function]}
        onKeyUp={[Function]}
        style={
          Object {
            "fontSize": "16px",
          }
        }
        tabIndex="0"
      >
        <span
          className="country-label"
        >
          Select a country
        </span>
        <span
          className="arrow-down "
        >
          â–¾
        </span>
      </div>
    </div>
  </nav>
</div>
`;
